File: simple-bank-cli-python/src/model/domain/wallet.py
class Wallet:

    def __init__(self, current_value):
        self.current_value = current_value
        self.check_not_negative()

    def get_value(self):
        return self.current_value

    def debit(self, value):
        self.current_value -= value

    def credit(self, value):
        self.current_value += value

    def check_not_negative(self):
        if (self.current_value < 0.0):
            raise ValueError('Current value invalid')

File: simplified-payment/ms-transference/src/main/java/br/com/challenge/payment/core/model/Wallet.java
package br.com.challenge.payment.core.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

import java.math.BigDecimal;

@Table("tb_wallet")
@Data
@NoArgsConstructor
public class Wallet {

    @Id
    private Integer id;

    private Double currentValue;

    public Wallet(Double currentValue) {
        this.currentValue = currentValue;
    }

    public void debit(Double value) {
        this.currentValue = BigDecimal.valueOf(value)
                .subtract(getCurrentValueAsBigDecimal())
                .doubleValue();
    }

    public void credit(Double value) {
        this.currentValue = BigDecimal.valueOf(value)
                .add(getCurrentValueAsBigDecimal())
                .doubleValue();
    }

    public BigDecimal getCurrentValueAsBigDecimal() {
        return BigDecimal.valueOf(currentValue);
    }

    public boolean isCanTransfer(Double value) {
        final int negative = -1;
        int signum = getCurrentValueAsBigDecimal()
                .subtract(BigDecimal.valueOf(value))
                .signum();
        return signum != negative;
    }
}
